storybookjs/storybook
# fix: controls missing when navigating from story

Closes #21805

## What I did

If a story has disabled the controls add-on and navigate to another story that hasn't disabled it, the add-on may still appear

Please note that this PR cause the components in the `Secondary` story to flicker(`Controls - Actions - Interactions` -> `Actions - Interactions`). I am not sure if this change is correct, so please let me know if there are any issues
## How to test

1. Run a sandbox for template, e.g. `yarn task --task sandbox --start-from auto --template react-vite/default-ts`
2. Please update the value of `Secondary` in the file `sandbox/react-vite-default-ts/src/stories/Button.stories.ts` with the follow code

```ts
export const Secondary: Story = {
  args: {
    label: 'Button',
  },
  parameters: {
    controls: {
      disable: true
    }
  }
};
```

3. Open Storybook in your browser
4. Access `Secondary` story, the `controls` add-on is disabled
5. Access `Primary` story, the `controls` add-on appear

## Checklist



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

      make sure to add the `ci:merged` or `ci:daily` GH label to it.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`
